# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/MediaEngineWebRTCAudio.cpp
# Commit: 59fb95311b12
# Full Hash: 59fb95311b12b332460753c7e063e8046288c1e2
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-02-17 21:32:26
# Description:
#   Bug 1692823 - Only end AudioInputProcessing if it has been set. r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D105447
# ==============================================================================

diff -r 15d9ff6377e1 -r 59fb95311b12 dom/media/webrtc/MediaEngineWebRTCAudio.cpp
--- a/dom/media/webrtc/MediaEngineWebRTCAudio.cpp	Wed Feb 17 13:44:38 2021 +0000
+++ b/dom/media/webrtc/MediaEngineWebRTCAudio.cpp	Wed Feb 17 14:02:42 2021 +0000
@@ -1262,7 +1262,9 @@
 
 void AudioInputTrack::DestroyImpl() {
   ProcessedMediaTrack::DestroyImpl();
-  mInputProcessing->End();
+  if (mInputProcessing) {
+    mInputProcessing->End();
+  }
 }
 
 void AudioInputTrack::ProcessInput(GraphTime aFrom, GraphTime aTo,
